src/main/webapp/resources/js/pages/projects/samples-metadata-import/redux/importReducer.ts

```diff
@@ -74,99 +79,101 @@ For more information on redux async thunks see: https://redux-toolkit.js.org/api
-              updateSample({
-                projectId,
-                sampleId,
-                body: {
-                  name,
-                  metadata: metadataFields,
-                },
-              })
-                .then(() => {
-                  metadataSaveDetails[index] = { saved: true };
-                })
-                .catch((error) => {
-                  metadataSaveDetails[index] = {
-                    saved: false,
-                    error: error.response.data.error,
-                  };
-                })
-            );
-          } else {
-            promises.push(
-              createSample({
-                projectId,
-                body: {
-                  name,
-                  metadata: metadataFields,
-                },
-              })
-                .then(() => {
-                  metadataSaveDetails[index] = { saved: true };
-                })
-                .catch((error) => {
-                  metadataSaveDetails[index] = {
-                    saved: false,
-                    error: error.response.data.error,
-                  };
-                })
-            );
-          }
-        }
-      }
-      await Promise.all(promises).then(() => {
-        dispatch(
-          setMetadataSaveDetails(Object.assign({}, metadataSaveDetails))
+      //save selected metadata entry rows
+      //during a partial failure only save data that has not already been saved
+      const selectedSampleList = metadata.filter((metadataItem) => {
+        const name: string = metadataItem[sampleNameColumn];
+        return (
+          selectedMetadataKeys.includes(metadataItem.rowKey) &&
+          metadataSaveDetails[name]?.saved !== true
         );
       });
-    }
 
-    return { metadataSaveDetails };
+      const createSampleList: CreateSampleItem[] = [];
+      const updateSampleList: UpdateSampleItem[] = [];
+
+      selectedSampleList.forEach((metadataItem) => {
+        const name = metadataItem[sampleNameColumn];
+        const sampleId = metadataValidateDetails[name].foundSampleId;
+        const metadata = createMetadataFields(
+          sampleNameColumn,
+          headers,
+          metadataItem
+        );
+
+        if (typeof sampleId === "number") {
+          updateSampleList.push({ name, sampleId, metadata });
+        } else {
+          createSampleList.push({ name, metadata });
+        }
+      });
+
+      const {
+        promiseList: createPromiseList,
+        newMetadataSaveDetails: createMetadataSaveDetails,
+      } = generatePromiseList(
+        createSampleList,
+        createSamples,
+        projectId,
+        selectedSampleList.length,
+        metadataSaveDetails,
+        dispatch
+      );
+      await Promise.all(createPromiseList);
+
+      const {
+        promiseList: updatePromiseList,
+        newMetadataSaveDetails: updateMetadataSaveDetails,
+      } = generatePromiseList(
+        updateSampleList,
+        updateSamples,
```

Comment:
Check your warnings in this file and on this line in particular.

Reply:
Please see c391d577909b3071b86b971f1b3bbca37fdf6ad2 and 3ce8e16404e51b0a097315f2311ba0e916dfec23.